sycl/doc/extensions/proposed/sycl_ext_intel_fpga_kernel_interface_properties.asciidoc

```diff
@@ -0,0 +1,213 @@
+
+== Kernel Properties
+
+```c++
+namespace sycl {
+namespace ext {
+namespace oneapi {
+namespace experimental {
+
+// Corresponds to streaming_kernel
+struct streaming_kernel_key {
+  using value_t = property_value<streaming_kernel_key>;
+}; // streaming_kernel_key
+
+// Corresponds to csr_kernel
+struct csr_kernel_key {
+  using value_t = property_value<csr_kernel_key>;
+}; // csr_kernel_key
+
+// Corresponds to stall_free_return_kernel
+struct stall_free_return_kernel_key {
+  using value_t = property_value<stall_free_return_kernel_key>;
+}; // stall_free_return_kernel_key
+
+// Corresponds to pipelined_kernel
+struct pipelined_kernel_key {
+  using value_t = property_value<pipelined_kernel_key>;
+}; // pipelined_kernel_key
+
+inline constexpr streaming_kernel_key::value_t streaming_kernel;
+
+inline constexpr csr_kernel_key::value_t csr_kernel;
+
+inline constexpr stall_free_return_kernel_key::value_t stall_free_return_kernel;
+
+inline constexpr pipelined_kernel_key::value_t pipelined_kernel;
+
+} // namespace experimental
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+```
+
+|===
+|Property|Description
+|`streaming_kernel`
+|The `streaming_kernel` property adds the requirement that the kernel must
+ have dedicated ports for input / output signals. This applies for both
+ control, and kernel argument data signals.
+
+|`csr_kernel`
+|The `csr_kernel` property adds the requirement that the kernel must have
+ its input / output control and kernel argument data signals placed in a shared
+ Control and Status Register (CSR) map.
+
+|`stall_free_return_kernel`
+|The `stall_free_return_kernel` property provides the compiler the guarantee
+ that downstream logic will not back-pressure the kernel.
+
+|`pipelined_kernel`
+| The `pipelined_kernel` property directs the compiler to pipeline calls to the
+ kernel so that multiple invocations of the kernel can be in flight
+ simultaneously.
+
+|===
+
+Device compilers that do not support this extension may accept and ignore these
+ properties.
+
+== Adding a Property List to a Kernel Launch
+
+A simple example of using this extension to launch a kernel with a streaming
+ interface is shown below.
+
+The example assumes that the kernel will not be back-pressured by the downstream
+logic and hence uses the property `stall_free_return_kernel`:
+
+```c++
+sycl::ext::oneapi::experimental::properties properties{
+  sycl::ext::oneapi::experimental::streaming_kernel,
+  sycl::ext::oneapi::experimental::stall_free_return_kernel};
+
+q.single_task(properties, [=]() {
+  *a = *b + *c;
+}).wait();
+```
+
+== Embedding Properties into a Kernel
+
+The example below shows how the kernel from the previous section could be
+rewritten to leverage an embedded property list:
+
+```c++
+struct KernelFunctor {
+
+  KernelFunctor(int* a, int* b, int* c) : a(a), b(b), c(c) {}
+
+  void operator()() const {
+    *a = *b + *c;
+  }
+
+  auto get(sycl::ext::oneapi::experimental::properties_tag) {
```

Comment:
Is this a magic function that encode the properties? How it is connected to the `single_task` below?